### PR TITLE
Fix remaining code coverage collection issues

### PR DESCRIPTION
**Pull request by coolreader18**

### CODE REVIEW

File: vm/src/stdlib/os.rs [hunk 1]
```diff
@@ -1274,7 +1274,7 @@ mod _os {
         Ok(())
     }
 
-    #[cfg(unix)]
+    #[cfg(all(unix, not(any(target_os = "redox", target_os = "android"))))]
```
Comment (fanninpm) [inline vm/src/stdlib/os.rs:1277]:
```suggestion
    #[cfg(all(unix, not(any(target_os in {"redox", "android"}))))]
```

**Comment by fanninpm** [comment]
Are there any plans to add coverage testing to PRs?

**Comment by coolreader18** [comment]
I think we used to have that a while ago, but since it's a 4 hour ci run now no, probably not

**Comment by coolreader18** [comment]
Yooo, nice!

**Comment by coolreader18** [comment]
Oooh wait, I've found a much better way to do code coverage

**Comment by coolreader18** [comment]
WOW, that's so much better than the other method of collection profiling data. It actually lets the compiler optimize the code, so we can run with `--release` no problem. [Article about source-based code coverage](https://blog.rust-lang.org/inside-rust/2020/11/12/source-based-code-coverage.html)

**Review by youknowone — APPROVED** [review]

**Pull request merged**